price-estimator/src/main.rs

```diff
@@ -1,34 +1,25 @@
+mod filter;
 mod orderbook;
 
 use core::{
     contracts::{stablex_contract::StableXContractImpl, web3_provider},
     http::HttpFactory,
     metrics::{HttpMetrics, MetricsServer},
     orderbook::EventBasedOrderbook,
-    orderbook::StableXOrderBookReading,
     util::FutureWaitExt as _,
 };
 use ethcontract::PrivateKey;
+use filter::TokenPair;
 use prometheus::Registry;
 use std::{num::ParseIntError, path::PathBuf, sync::Arc, thread, time::Duration};
 use structopt::StructOpt;
 use tokio::{runtime, time};
 use url::Url;
-use warp::Filter;
+use warp::{Filter, Rejection};
 
 #[derive(Debug, StructOpt)]
 #[structopt(name = "price estimator", rename_all = "kebab")]
 struct Options {
-    /// The log filter to use.
-    ///
-    /// This follows the envlogger syntax (e.g. 'info,driver=debug').
-    #[structopt(
-        long,
-        env = "LOG_FILTER",
-        default_value = "warn,driver=info,price_estimator=info"
-    )]
-    log_filter: String,
-
```

Comment:
This wasn't being used because we use env_logger directly without going through slog.

Reply:
Seems reasonable.